packages/grid/src/vaadin-grid.js

```diff
@@ -561,6 +573,20 @@ class Grid extends ElementMixin(
           const cellWidth = c.offsetWidth + 1;
           col._currentWidth = Math.max(col._currentWidth, cellWidth);
         });
+
+      if (col.parentElement != null && col.parentElement.localName === 'vaadin-grid-column-group') {
+        const columnGroup = col.parentElement;
+
+        if (columnGroup._headerCell != null) {
+          const cellWidth = columnGroup._headerCell.offsetWidth + 1;
+          col._currentWidth = Math.max(col._currentWidth, cellWidth);
```

Comment:
Not sure if [this](https://github.com/vaadin/web-components/issues/483#issuecomment-831873153) is a requirement for this fix (ping @rolfsmeds): "in the case where the group header's width is larger than that of the columns within it, that the excess width is divided amongst the columns proportionally to their intrinsic widths, similarly to flex-grow"

Currently, seems that each child column adopts the full width of their parent group, even if there are multiple child columns in the group

```html
<vaadin-grid theme="column-borders">
  <vaadin-grid-column-group header="foo foo foo foo foo foo foo foo foo">
    <vaadin-grid-column header="foo" auto-width flex-grow="0"></vaadin-grid-column>
    <vaadin-grid-column header="foo foo foo" auto-width flex-grow="0"></vaadin-grid-column>
  </vaadin-grid-column-group>
</vaadin-grid>
```
<img width="617" alt="Screenshot 2021-11-02 at 11 31 00" src="https://user-images.githubusercontent.com/1222264/139821372-483bd473-4317-4f1c-92a8-c56d4f03efd6.png">

Reply:
Yeah that seems simply wrong. Proposed behavior would definitely be preferrable.